server/src/main/java/org/elasticsearch/index/mapper/MapperService.java

```diff
@@ -389,64 +400,62 @@ private String resolveDocumentType(String type) {
     /**
      * Returns the document mapper for this MapperService.  If no mapper exists,
      * creates one and returns that.
+     * @deprecated Get a {@link #snapshot} and call {@link Snapshot#documentMapperWithAutoCreate} on
+     *             it as many times as needed.
      */
+    @Deprecated
```

Comment:
I feel like it'd be useful to think about all the callers, yeah. And I think if something really does want to constant get fresh snapshots it can have a `MapperService` and call `snapshot` over and over again.

I don't have to deprecate the methods but I think it's clearer when you read them from the snapshot. I'd feel better if we only had one method hiding a volatile read rather than a bunch of them.